app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -220,9 +222,9 @@ def consumer_create
     def consumer_destroy
       User.as_anonymous_admin do
         if Setting['unregister_delete_host']
-          sync_task(::Actions::Katello::Host::Destroy, @host)
+          Katello::RegistrationManager.unregister_host(@host, :unregistering => false)
         else
-          sync_task(::Actions::Katello::Host::Unregister, @host)
+          Katello::RegistrationManager.unregister_host(@host, :unregistering => true)
```

Comment:
nit pick, you could just say:   Katello::RegistrationManager.unregister_host(@host, :unregistering => !Setting['unregister_delete_host'])